DataFormats/PatCandidates/src/TauPFEssential.cc

```diff
@@ -26,3 +26,28 @@ pat::tau::TauPFEssential::TauPFEssential(const reco::PFTau& tau) :
     p4Jet_ = tau.jetRef()->p4();
   }
 }
+
+pat::tau::TauPFEssential::TauPFEssential(const reco::PFBaseTau& tau) :
+    p4Jet_(reco::Candidate::LorentzVector()),
+    p4CorrJet_(reco::Candidate::LorentzVector()),
+    decayMode_(tau.decayMode()),
+    dxy_(0.),
+    dxy_error_(1.e+3),
+    hasSV_(false),
+    ip3d_(0.),
+    ip3d_error_(1.e+3),
+    ecalEnergy_(0.),
+    hcalEnergy_(0.),
+    leadingTrackNormChi2_(1.e+3),
+    phiAtEcalEntrance_(0.),
+    etaAtEcalEntrance_(0.),
+    ecalEnergyLeadChargedHadrCand_(0.),
+    hcalEnergyLeadChargedHadrCand_(0.),
+    etaAtEcalEntranceLeadChargedCand_(0.),
+    ptLeadChargedCand_(0.),
+    emFraction_(0.)
+{
+  if ( tau.jetRef().isAvailable() && tau.jetRef().isNonnull() ) { // CV: add protection to ease transition to new CMSSW 4_2_x RecoTauTags
```

Comment:
given that the transition to CMSSW_4_2_X has happened long time ago, is this "if" still needed?